### PR TITLE
Cleanups part 2

### PR DESCRIPTION
**Pull request by JustinDrake**
Changelog (nothing major):

* Clean up constants
   * Add to table of contents
   * Rename a few
   * Split into more categories (e.g. "Deposit contract"  and "Initial values")
   * Replaced `SQRT_E_DROP_TIME ` by `INACTIVITY_PENALTY_QUOTIENT`
* Put all the data structures in "Data structures", including specials
* Move deposit contract after "Data structures"
* `ShardAndCommittee` => `ShardCommittee` (and related changes)
* `aggregate_sig` => `aggregate_signature`
* `justified_slot_bitfield` => `justification_bitfield`
* Use double-ticks everywhere in Python docstrings
* Clean up deposit contract a bit

### CODE REVIEW

**Comment by djrtwo** [comment]
Request that you don't add much more to this PR. The last cleanup PR took a _very_ long time to review due to the amount of content/changes. I understand if this is WIP for the given changes but please try to keep the scope reasonable.

**Comment by JustinDrake** [comment]
@djrtwo Ok :) I will add a changelog and leave it as is.

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -701,17 +807,17 @@ Here's a diagram of what is going on:
...
 ![](http://vitalik.ca/files/ShuffleAndAssign.png?1)
 
-#### `get_shard_and_committees_for_slot`
+#### `get_shard_committees_at_slot`
 
 ```python
-def get_shard_and_committees_for_slot(state: BeaconState,
-                                      slot: int) -> List[ShardAndCommittee]:
+def get_shard_committees_at_slot(state: BeaconState,
+                                      slot: int) -> List[ShardCommittee]:
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: indent
```suggestion
                                 slot: int) -> List[ShardCommittee]:
```

File: specs/core/0_beacon-chain.md [hunk 2]
```diff
@@ -135,29 +164,29 @@ Unless otherwise indicated, code appearing in `this style` is to be interpreted
...
-**Validator status codes**
+### Status codes
 
 | Name | Value |
 | - | - |
 | `PENDING_ACTIVATION` | `0` |
 | `ACTIVE` | `1` |
-| `EXITED_WITHOUT_PENALTY` | `2` |
-| `EXITED_WITH_PENALTY` | `3` |
+| `PENDING_EXIT | `2` |
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: Note: fixes #195

File: specs/core/0_beacon-chain.md [hunk 3]
```diff
@@ -185,72 +214,9 @@ Unless otherwise indicated, code appearing in `this style` is to be interpreted
 **Notes**
 
 * See a recommended min committee size of 111 [here](https://vitalik.ca/files/Ithaca201807_Sharding.pdf); the shuffling algorithm will generally ensure the committee size is at least half the target.
-* The `SQRT_E_DROP_TIME` constant is the amount of time it takes for the inactivity leak to cut deposits of non-participating [validators](#dfn-validator) by ~39.4%.
+* The `INACTIVITY_PENALTY_QUOTIENT` equals `SQRT_E_DROP_TIME**2` where `SQRT_E_DROP_TIME := 2**17 slots` (~9 days) is the amount of time it takes for the inactivity penalty to cut deposits of non-participating [validators](#dfn-validator) by ~39.4%. The portion lost by offline [validators](#dfn-validator) after `D` epochs is about `D*D/2/INACTIVITY_PENALTY_QUOTIENT`.
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: I feel it's not very coherent when I read this. Maybe move the description of `INACTIVITY_PENALTY_QUOTIENT` to "Constants - Reward and penalty quotients" section?

File: specs/core/0_beacon-chain.md [hunk 4]
```diff
@@ -685,7 +791,7 @@ def get_new_shuffling(seed: Hash32,
         shard_id_start = crosslinking_start_shard + slot * committees_per_slot
 
         shards_and_committees_for_slot = [
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]:
```suggestion
        shard_committees = [
```

File: specs/core/0_beacon-chain.md [hunk 5]
```diff
@@ -750,22 +856,25 @@ def get_attestation_participants(state: State,
...
-            participants.append(validator_index)
-    return participants
+
+    # Find the relevant committee
+    shard_committees = get_shard_committees_at_slot(state, attestation_data.slot)
+    shard_committee = [x for x in shard_committees if x.shard == attestation_data.shard][0]
+    assert len(participation_bitfield) == ceil_div8(len(shard_committee.committee))
+
+    # Find the attesters in the committee
+    attesters = []
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: IMO I think it's fine to name it `participants` here. "Attester" is their job/role, but only the participated ones will be added to this list.

File: specs/core/0_beacon-chain.md [hunk 6]
```diff
@@ -135,29 +164,29 @@ Unless otherwise indicated, code appearing in `this style` is to be interpreted
 | `MIN_VALIDATOR_REGISTRY_CHANGE_INTERVAL` | `2**8` (= 256) | slots | 25.6 minutes |
 | `POW_RECEIPT_ROOT_VOTING_PERIOD` | `2**10` (= 1,024) | slots | ~1.7 hours |
 | `SHARD_PERSISTENT_COMMITTEE_CHANGE_PERIOD` | `2**17` (= 131,072) | slots | ~9 days |
-| `SQRT_E_DROP_TIME` | `2**17` (= 131,072) | slots | ~9 days |
 | `COLLECTIVE_PENALTY_CALCULATION_PERIOD` | `2**20` (= 1,048,576) | slots | ~73 days |
-| `DELETION_PERIOD` | `2**22` (= 16,777,216) | slots | ~290 days |
+| `EMPTY_VALIDATOR_EXPIRY` | `2**22` (= 16,777,216) | slots | ~290 days |
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: Would "expiry" sound more like the "timestamp"? What do you think about `EMPTY_VALIDATOR_EXPIRY_PERIOD`?

File: specs/core/0_beacon-chain.md (hunk 3 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: Right. Each constant section can have their own notes 👍Will fix.

File: specs/core/0_beacon-chain.md (hunk 5 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: Ok 👍

File: specs/core/0_beacon-chain.md (hunk 6 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: Naming is hard :) This constant refers to the amount of time a validator entry with a balance of zero stays in the registry before being cleaned up. What about `ZERO_BALANCE_VALIDATOR_TTL`?

File: specs/core/0_beacon-chain.md [hunk 7]
```diff
@@ -498,6 +466,144 @@ When the contract publishes a `ChainStart` log, this initializes the chain, call
...
+
+```python
+MIN_DEPOSIT: constant(uint256) = 1  # ETH
+MAX_DEPOSIT: constant(uint256) = 32  # ETH
+GWEI_PER_ETH: constant(uint256) = 1000000000  # 10**9
+CHAIN_START_FULL_DEPOSIT_THRESHOLD: constant(uint256) = 16384  # 2**14
+DEPOSIT_CONTRACT_TREE_DEPTH: constant(uint256) = 32
+SECONDS_PER_DAY: constant(uint256) = 86400
+
+DepositEvent: event({previous_receipt_root: bytes32, data: bytes[2064], full_deposit_count: uint256})
```
Comment (djrtwo) [inline specs/core/0_beacon-chain.md]: `Event` is redundant.

Consider just calling it `Deposit`. It is a `Deposit` event

**Comment by djrtwo** [comment]
solid work @JustinDrake

File: specs/core/0_beacon-chain.md [hunk 8]
```diff
@@ -498,6 +466,144 @@ When the contract publishes a `ChainStart` log, this initializes the chain, call
...
+```python
+{
+    # Receipt Merkle branch
+    'merkle_branch': '[hash32]',
+    # Merkle tree index
+    'merkle_tree_index': 'uint64',
+    # Deposit data
+    'deposit_data': {
+        'deposit_parameters': DepositParametersRecord,
+        'value': 'uint64',
```
Comment (prestonvanloon) [inline specs/core/0_beacon-chain.md]: Unit? Is this gwei?

File: specs/core/0_beacon-chain.md (hunk 6 above)
Comment (hwwhww) [reply specs/core/0_beacon-chain.md]: `ZERO_BALANCE_VALIDATOR_TTL` sounds good to me.

File: specs/core/0_beacon-chain.md (hunk 8 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: Yes, is Gwei. I'll be explicit in the comment.

**Pull request merged**